tests/with_thiserror_tests.rs

```diff
@@ -68,3 +68,31 @@ fn test_errordoc_for_struct() {
     }
     assert_display(MyError { variant: 42 }, "I'm a doc for MyError");
 }
+
```

Comment:
> So this is only partially complete. This covers the case where an error is implemented as a tuple, but thiserror also supports implying the source via a field named source on struct variants, and it supports manually marking a field as the source, so we need to add test cases for the following struct definitions:

I have added tests here to see the case you mentioned is working.
It's working.